### ISSUE TITLE
`no-extra-parens` with async IIFE

### ISSUE THREAD

**Issue by hzoo** [babel/eslint-plugin-babel#19]
```js
(async function () {

}());
```

`Gratuitous parentheses around expression no-extra-parens`

Ref https://github.com/babel/babel-eslint/issues/186

**Comment by bathos** [comment]
Agreed that this is a bug, though I would point out that in a sense, it is (perhaps unintentionally) accurate; the parens could be considered gratuitous because, being immediately invoked, the function is already not relying on a specific contextual binding, and thus an arrow function (which is an expression already by definition) can always be used in its place, paren-free:

```js
async () => {

}();
```
